src/Compilers/Shared/BuildClient.cs

```diff
@@ -217,11 +269,57 @@ public Task<RunCompilationResult> RunCompilationAsync(IEnumerable<string> origin
             }
         }
 
-        protected abstract Task<BuildResponse> RunServerCompilation(List<string> arguments, BuildPaths buildPaths, string sessionName, string keepAlive, string libDirectory, CancellationToken cancellationToken);
+        private Task<BuildResponse> RunServerCompilation(
+            List<string> arguments,
+            BuildPaths buildPaths,
+            string sessionKey,
+            string keepAlive,
+            string libDirectory,
+            CancellationToken cancellationToken)
+        {
+            return RunServerCompilationCore(_language, arguments, buildPaths, sessionKey, keepAlive, libDirectory, _timeoutOverride, _createServerFunc, cancellationToken);
+        }
+
+        private static Task<BuildResponse> RunServerCompilationCore(
+            RequestLanguage language,
+            List<string> arguments,
+            BuildPaths buildPaths,
+            string pipeName,
+            string keepAlive,
+            string libEnvVariable,
+            int? timeoutOverride,
+            CreateServerFunc createServerFunc,
+            CancellationToken cancellationToken)
+        {
+            var alt = new BuildPathsAlt(
```

Comment:
>alt [](start = 16, length = 3)

Why can't we use `buildPaths` directly in the call to `RunServerCompilationCore`?